Lib/test/test_generators.py

```diff
@@ -270,6 +270,21 @@ def gen():
         self.assertEqual(next(g), "done")
         self.assertEqual(sys.exc_info(), (None, None, None))
 
+    def test_except_throw_bad_exception(self):
+        class E(Exception):
+            def __new__(cls, *args, **kwargs):
+                return cls
+
+        def boring_generator():
+            yield
+
+        gen = boring_generator()
+
+        with self.assertRaisesRegex(
+                TypeError,
+                'should have returned an instance of BaseException'):
+            gen.throw(E)
```

Comment:
It would be interesting to check if the generator was resumed or not here. Can you add `assert next(gen) is None` after the `with` to prove that it has not been resumed? (or perhaps `yield 42` and assert that for clarity)

Reply:
(if it is resumed here, then test for StopIteration